tests/app/celery/test_tasks.py

```diff
@@ -861,26 +872,44 @@ def test_process_rows_sends_save_task(
                 "to": "recip",
                 "row_number": "row_num",
                 "personalisation": {"foo": "bar"},
-                "queue": None,
+                "queue": "send-{}-tasks".format(template_type),
                 "client_reference": reference,
                 "sender_id": str(sender_id) if sender_id else None,
             }
         )
         task_mock.apply_async.assert_called_once()
 
     @pytest.mark.parametrize(
-        "csv_threshold, expected_queue",
+        "csv_normal_threshold, csv_bulk_threshold, template_process_type, expected_queue",
         [
-            (0, "bulk-tasks"),
-            (1_000, None),
+            (1, 1_000, PRIORITY, "priority-tasks"),  # keep priority task even if normal threshold is met
+            (2, 1_000, PRIORITY, "priority-tasks"),  # keep priority when no thresholds are met
+            (0, 1, PRIORITY, "bulk-tasks"),  # autoswitch to bulk queue if bulk threshold is met, even if in priority.
+            (1, 1, PRIORITY, "bulk-tasks"),  # autoswitch to bulk queue if bulk threshold is met.
+            (1, 1, NORMAL, "bulk-tasks"),  # autoswitch to bulk queue if bulk threshold is met.
+            (1, 1_000, NORMAL, "send-email-tasks"),  # keep normal priority
+            (2, 1_000, NORMAL, "send-email-tasks"),  # keep normal priority
+            (1_000, 1_000, NORMAL, "send-email-tasks"),  # keep normal priority
+            (1, 1, BULK, "bulk-tasks"),  # keep bulk priority
+            (1, 1_000, BULK, "send-email-tasks"),  # autoswitch to normal queue if normal threshold is met.
+            (2, 1_000, BULK, "bulk-tasks"),  # keep bulk priority
```

Comment:
So if they're set to `BULK` and send only 1 email (ie less than the normal threshold) they'll stay as bulk? But if they sent more to meet the threshold they'd be normal. That seems odd. Shouldn't all `bulk` csvs that are below the bulk threshold get sent as normal?

Reply:
oh yes you might be right, let me review this